1단계 첫번째 PR

gaeul-kim opened the pull request:
단위테스트를 만들고 적용하는게 많이 어렵네요.
일단 나름의 생각대로 구현해 보았는데, 어떤 방식으로 접근하면 더 좋을지 조언 부탁드립니다!

Commit:
first commit

javajigi commented on src/main/java/racingcar/Car.java:
```diff
@@ -0,0 +1,28 @@
+ */
+public class Car {
+    Integer position;
+    MoveScore moveScore;
```
인스턴스 변수는 특별한 이유는 없는 한 private으로 구현한다.

javajigi commented on src/main/java/racingcar/MoveScore.java:
```diff
@@ -0,0 +1,18 @@
+ * @author sangsik.kim
+ */
+public class MoveScore {
+    Integer LIMIT_SCORE = 10;
```
상수 값 convention을 따라라.
상수 값은 static final로 구현하면 변수 이름을 모두 대문자로 구현하는 것이 관례이다.

javajigi commented on src/main/java/racingcar/MoveScore.java:
```diff
@@ -0,0 +1,18 @@
+    Integer LIMIT_SCORE = 10;
+
+    public Boolean canMove() {
+        return generateRandomMoveScore() >= 4;
```
값을 하드코딩하지 마라.
상수(static final)를 만들고 이름을 부여해 이 변수의 역할이 무엇인지 의도를 드러내라.

javajigi commented on src/main/java/racingcar/Car.java:
```diff
@@ -0,0 +1,28 @@
+        this.moveScore = new MoveScore();
+    }
+
+    public void move() {
```
MoveScore를 move() 메소드의 인자로 전달하는 것은 어떨까?
지금 구현한 코드와 인자로 전달했을 때의 장,단점은 무엇이 있을까?

javajigi commented on src/main/java/racingcar/RacingGame.java:
```diff
@@ -0,0 +1,74 @@
+ */
+public class RacingGame {
+    List<Car> cars;
+    Integer time;
```
인스턴스 변수는 특별한 이유는 없는 한 private으로 구현한다.

javajigi replied:
time을 인스턴스 변수로 만들지 말고 구현해 본다.

javajigi commented on src/main/java/racingcar/RacingGame.java:
```diff
@@ -0,0 +1,74 @@
+        for (int i = 0; i < position; i++) {
+            result.append("-");
+        }
+        return result.toString();
```
핵심 로직을 구현하는 부분(상태 변경 등과 같은)과 View 로직(System.in을 통해 값 입력, System.out을 통해 값 출력)을 분리한다.
한 객체에 이 두 가지 로직이 같이 구현되어 있는 경우 추후 View가 되었을 때 핵심 로직을 구현하는 코드를 재사용할 수 없다.
또한 View에 종속되는 순간 테스트하기도 힘들다. 
이 연습은 웹 프로그래밍을 할 때 DB에 종속된 코드와 DB에 종속되지 않은 코드를 분리함으로써 테스트 가능하도록 구현하는 것과 같은 연습이다.

javajigi commented on src/main/java/racingcar/RacingGame.java:
```diff
@@ -0,0 +1,74 @@
+        return result.toString();
+    }
+
+
```
불필요하게 공백 라인을 만들지 않는다.
공백 라인을 뛰우는 것도 코드상에 문맥이 달라지는 부분에 의도를 가지고 뛰우면 좋겠다.

javajigi commented on src/test/java/racingcar/RacingGameTest.java:
```diff
@@ -0,0 +1,35 @@
+        RacingGame game = new RacingGame();
+        game.start(3, 5);
+    }
+
```
중요한 로직이라 할 수 있는 Car가 이동할 때와 이동하지 않을 때에 대한 Test Case가 없네요.

javajigi reviewed (changes requested):
구현 잘 했네요. 피드백 남겼어요.

gaeul-kim committed:
# second commit

각자의 역할만을 수행할 수 있도록 최대한 분리
어쩔 수 없이 의존관계를 강하게 맺는 경우가 발생
인터페이스를 좀더 활용하면 좋을듯

### convention 수정
- 인스턴스 변수 접근제어자 private 변경
- 하드코딩 되어있는 매직넘버 상수로 변경
- 상수 값 final static 사용
- 불필요한 공백 제거

### 구현
- 테스트 케이스 추가
- View 로직 분리
- 확장 가능성이 있는부분은 인터페이스 사용
